docs/advanced_topics/streamfield_migrations.md

```diff
@@ -0,0 +1,566 @@
+                [{
+                    "value": body,
+                    "type": "rich_text"
+                }],
+                cls=DjangoJSONEncoder)
+            changed = True
+        else:
+            # It's already valid JSON. Leave it.
+            pass
+    return revision_data, changed
+
+
+def page_to_richtext(page):
+    changed = False
+    if page.body.raw_text is None:
+        raw_text = ''.join([
+            child.value.source for child in page.body
+            if child.block_type == 'rich_text'
+        ])
+        page.body = raw_text
+        changed = True
+    return page, changed
+
+
+def pagerevision_to_richtext(revision_data):
+    changed = False
+    body = revision_data.get('body', 'definitely non-JSON string')
+    if body:
+        try:
+            body_data = json.loads(body)
+        except ValueError:
+            # It's not apparently a StreamField. Leave it.
+            pass
+        else:
+            raw_text = ''.join([
+                child['value'] for child in body_data
+                if child['type'] == 'rich_text'
+            ])
+            revision_data['body'] = raw_text
+            changed = True
+    return revision_data, changed
+
+
+def convert(apps, schema_editor, page_converter, pagerevision_converter):
+    BlogPage = apps.get_model("demo", "BlogPage")
+    for page in BlogPage.objects.all():
+
+        page, changed = page_converter(page)
+        if changed:
+            page.save()
+
+        for revision in page.revisions.all():
+            revision_data = revision.content
+            revision_data, changed = pagerevision_converter(revision_data)
+            if changed:
+                revision.content = revision_data
+                revision.save()
+
+
+def convert_to_streamfield(apps, schema_editor):
+    return convert(apps, schema_editor, page_to_streamfield, pagerevision_to_streamfield)
+
+
+def convert_to_richtext(apps, schema_editor):
+    return convert(apps, schema_editor, page_to_richtext, pagerevision_to_richtext)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        # leave the dependency line from the generated migration intact!
+        ('demo', '0001_initial'),
+    ]
+
+    operations = [
+        # leave the generated AlterField intact!
+        migrations.AlterField(
+            model_name='BlogPage',
+            name='body',
+            field=wagtail.fields.StreamField([('rich_text', wagtail.blocks.RichTextBlock())]),
+        ),
+
+        migrations.RunPython(
+            convert_to_streamfield,
+            convert_to_richtext,
+        ),
+    ]
+```
+
+## StreamField data migrations
+
+As of version 4.2, Wagtail includes code from the [wagtail-streamfield-migration-toolkit](https://github.com/wagtail/wagtail-streamfield-migration-toolkit) package, which was created as part of Wagtail's participation in Google Summer of Code, 2022. The migration utilities from this package are exposed through:
```

Comment:
Other than a `versionadded` note (which we can systematically search for and drop at the appropriate time), we should avoid talking about how and when a feature was introduced - that's not relevant information to someone who just wants to use it. Anyone who is interested in changes to Wagtail over time can look at the release notes.